### DISCUSSION TITLE
Playing with ChatGPT

### DISCUSSION THREAD

**Discussion by johnbent** [tekinged/missing#448] [General]
So I paid $20 to try ChatGPT 4.0 for a month. I know most, if not all, of the transformation rules are defined in Josephs but I have a hard time reading Josephs for some reason. So I asked ChatGPT to derive the rules for the transformation from a base verb to the resulting state verb (e.g. omart -> blart).

It gave me a set of rules which don't make sense:
`Sure, let's analyze the given list to identify some patterns for converting base verbs into their resulting state forms in Palauan. This is a preliminary observation, and my analysis might not capture all the nuances of Palauan verb morphology.

Here are some observed patterns:

Pattern 1: For base verbs that start with 'm', the resulting state form replaces 'm' with 's' or 'se'.

Example: mesilek (base: mesilek), selilik (resulting state: selilik)
Pattern 2: For base verbs that start with 'm', sometimes 'm' is replaced by 'ch'.

Example: mengertall (base: mengertall), kerretall (resulting state: kerretall)
Pattern 3: For base verbs that start with 'm' followed by a vowel, the 'm' is often replaced by 'd'.

Example: meleb (base: meleb), delebongel (resulting state: delebongel)
Pattern 4: For base verbs that start with 'o', the 'o' is often replaced with 'bl' or 'ul' or 'b'.

Example: omatek (base: omatek), blatek (resulting state: blatek)
Pattern 5: For base verbs that start with 'o' followed by another vowel, the 'o' is replaced by 'u' and the next vowel.

Example: olsiu (base: olsiu), ulsiaol (resulting state: ulsiaol)
Pattern 6: For base verbs that start with 'me', the 'me' is often replaced by 'te', 'de' or 'ng'.

Example: meluked (base: meluked), ngluked (resulting state: ngluked)
Pattern 7: The base and resulting forms appear to be quite different sometimes, suggesting more complex morphological changes or exceptions.

Example: melasech (base: melasech), delasech (resulting state: delasech)`

But then I asked it to guess the resulting state forms for 9 verbs that I hadn't given it in the training set. It did pretty well:

mengeed: I'd guess "chengeed" (based on 'm' -> 'ch') Wrong. Should be cheleed.
omechederiib: I'd guess "blachederiib" (based on 'o' -> 'bl') Wrong. Should be ulechederiib.
omtebechel: I'd guess "bltebechel" (based on 'o' -> 'bl')  Wrong. Should be ultebechel.
oltekau: I'd guess "ultekau" (based on 'o' -> 'ul') CORRECT!
mengull: I'd guess "chelull" (based on 'm' -> 'ch') Wrong. Should be klull
omad: I'd guess "blad" (based on 'o' -> 'bl') CORRECT!
meririau: I'd guess "rririau" (based on 'me' -> 'r') CORRECT!
mengeolt: I'd guess "kleolt" (based on 'm' -> 'kl') CORRECT!
omarech: I'd guess "blarech" (based on 'o' -> 'bl') CORRECT!
Again, these are based on preliminary observations and may not be accurate. Feel free to correct me!

So, with a training set of about 900 pairs, ChatGPT was able to correctly predict the resulting state form for 5 out of 9.

Might have some potential!

**Comment by johnbent** [comment]
Here's how it did with anticipatory state transformations:

orreteret - anticipatory form: tetetel
mesmengt - anticipatory form: semengt
omekang - anticipatory form: ukekeng
omades - anticipatory form: udades
melasem - anticipatory form: tesesem
oltak - anticipatory form: utak
melechang - anticipatory form: techechell
omerius - anticipatory form: ururius
merous - anticipatory form: rerus
melemotem - anticipatory form: temotem

It got all 10 wrong. Bummer.
orreteret	ortertall
mesmengt	smengtall
omekang	okall
omades	bedesall
melasem	esemall
oltak	otekiall
melechang	lechengaol
omerius	berusall
merous	ruikl
melemotem	temetamel

**Reply by johnbent** [reply]
It didn't do well on possessed nouns either unfortunately.

Although once I explained vowel weakening to it, it did somewhat better:

biich bichel bichel [CORRECT]
kak kekel kekengel [wrong]
ongeromel ongeremelel ongermelel [wrong]
kuoku kekuel kuekuel [wrong]
udai udeel udeul [wrong]
dereborb dereberbel dereberebel [wrong]
omecheroll omecherelel omecherellel [wrong]
reched rechedel rechedel [CORRECT]
ngebard ngeberdel ngeberdel [CORRECT]
molech melechel melechel [CORRECT]
olechukl olecheklel olecheklel [CORRECT]
riid reedel ridel [wrong]
chetakl cheteklel cheteklel [CORRECT]
olalem olalemel ollemel [wrong]
bai beel bil [wrong]
tabek tebekel tebekel [CORRECT]

7 out of 16. Not so great still.

**Comment by Lukes-Imeyuns** [comment]
Wow. Very interesting. I'm still quite hesitant about ChatGPT though, like
a lot. I prefer humans. 😅

So let me take this opportunity to say Thank You to you, John, and everyone
for all the work they put in THIS. I don't think I've ever said that...

I appreciate and enjoy that I'm a part of this collaboration and I love
that it's given me an opportunity to learn my own language more.

Ko meral mesaul el rokui!

Lukes

**Comment by johnbent** [comment]
It really is very impressive. I discovered some annoying warnings in some of the tekinged results. Instead of fixing it myself, I had this conversation:
1. Hey, do you understand this php warning: PASTED THING
2. Yeah, I do. Let me tell you a lot about it. Blah blah blah ..... [I interrupted it]
3. Nevermind that. I'm gonna paste you the offending function and you just rewrite it for me please.
4. Sure.
5. [Giant paste of code that was producing annoying warnings.]
6. [replied with fixed version]

I copied the fixed version and it worked perfectly. 30 minutes of work was done in 3.
